core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -0,0 +1,457 @@
+          throw new IllegalArgumentException("Column name: " + columns[0] + " in 'Column-Label Mapping' does not " +
+                                               "match the columns in the 'Column To Be Extracted' input text box. " +
+                                               "It has to be one of the columns in 'Column To Be Extracted' " +
+                                               "input text box.");
+        }
+        columnMapping.put(columns[0], columns[1]);
+      }
+    }
+
+    if (!Strings.isNullOrEmpty(excelInputreaderConfig.outputSchema)) {
+      String[] schemaList = excelInputreaderConfig.outputSchema.split(",");
+      for (String schema : schemaList) {
+        String[] columns = schema.split(":");
+        if (CollectionUtils.isNotEmpty(inputColumns) && !inputColumns.contains(columns[0])) {
+          throw new IllegalArgumentException("Column name: " + columns[0] + " in 'Field Name Schema Type Mapping'" +
+                                               " does not match the columns in the 'Column To Be Extracted' input " +
+                                               "text box. It has to be one of the columns in " +
+                                               "'Column To Be Extracted' input text box.");
+        }
+        outputSchemaMapping.put(columns[0], columns[1]);
+      }
+    }
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Object> input, Emitter<StructuredRecord> emitter) throws Exception {
+
+    getOutputSchema();
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    String inputValue = input.getValue().toString();
+
+    String[] excelRecords = inputValue.split("\t");
+
+    String fileName = excelRecords[1];
+    String sheetName = excelRecords[2];
+    String ifEndRow = excelRecords[3];
+
+    int prevRowNum = Integer.parseInt(excelRecords[0]);
+
+    if (filePrevRowNumMap.containsKey(fileName)) {
+      if (prevRowNum - filePrevRowNumMap.get(fileName) > 1 &&
+        excelInputreaderConfig.terminateIfEmptyRow.equalsIgnoreCase("true")) {
+        throw new ExecutionException("Encountered empty row while reading Excel file :" + fileName +
+                                       " . Terminating processing", new Throwable());
+      }
+    }
+
+    filePrevRowNumMap.put(fileName, prevRowNum);
+
+    Map<String, String> excelColumnValueMap = new HashMap<>();
+
+    for (String columns : excelRecords) {
+      String[] columnValueArray = columns.split("\r");
+      if (columnValueArray.length > 1) {
+        String columnName = columnValueArray[0];
+        String columnValue = columnValueArray[1];
+
+        if (columnMapping.containsKey(columnName)) {
+          excelColumnValueMap.put(columnMapping.get(columnName), columnValue);
+        } else {
+          excelColumnValueMap.put(columnName, columnValue);
+        }
+      }
+    }
+
+    try {
+      for (Schema.Field field : outputSchema.getFields()) {
```

Comment:
Ok! this logic looks totally wrong to me. Here is why : You are iterating over excelRecords. I am assuming excelRecords has more than 1 record (row) and the excelColumnValueMap gets overwritten everytime when it iterates through the excelRecords. Now, you take that and write to StructureRecord down here. So, it will write only the last record value to the output. This section needs to be within excelRecords loop. If all this makes sense, please re-write this section.